packages/uikit/src/molecules/Select/index.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useRef } from 'react';
+
+import Input from './Input';
+import Menu from './Menu';
+import { StickyContainer, PortalComponent } from '../../util/index';
+import { useOutsideClick, useKeyDown } from '../../hooks';
+
+interface SelectProps {
+  placeholder: string;
+  children?: any;
+  defaultValue?: any;
+}
+
+const Select: React.SFC<SelectProps> = ({
+  placeholder,
+  children,
+  defaultValue
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const ref = useRef<any>();
+  const [value, setValue] = useState<string>('');
+
+  useOutsideClick(ref, () => {
+    setIsOpen(false);
+  });
+
+  useKeyDown('Escape', () => {
+    setIsOpen(false);
+  });
+
+  const onSelect = (value: any) => {
+    setValue(value);
+    setIsOpen(false);
+  };
+
+  return (
+    <>
+      <Input
+        innerRef={ref}
+        isOpen={isOpen}
+        value={value}
+        placeholder={placeholder}
+        onClick={() => setIsOpen(!isOpen)}
+      />
+      {isOpen && (
+        <PortalComponent dom={document.body}>
+          <StickyContainer stickToEl={ref.current}>
+            <Menu depth={3}>{children({ onSelect })}</Menu>
+          </StickyContainer>
+        </PortalComponent>
+      )}
+    </>
+  );
+};
+
+export { default as Option } from './Option';
+
+export default Select;
```

Comment:
I wonder if `Option` is a descriptive enough name?
Maybe `Select.Option` would be nice? But not sure how to do that on functional components, only know how it works on classes 🤔